src/fu-history.c

```diff
@@ -1283,6 +1313,44 @@ fu_history_add_blocked_firmware(FuHistory *self, const gchar *checksum, GError *
 	return fu_history_stmt_exec(self, stmt, NULL, error);
 }
 
+gboolean
+fu_history_add_security_attribute(FuHistory *self,
+				  gchar *security_attr_json,
+				  gchar *hsi_score,
+				  GError **error)
+{
+	gint rc;
+	g_autoptr(sqlite3_stmt) stmt = NULL;
+	g_autoptr(GRWLockWriterLocker) locker = NULL;
+
+	g_return_val_if_fail(FU_IS_HISTORY(self), FALSE);
+
+	/* lazy load */
+	if (!fu_history_load(self, error))
+		return FALSE;
+	/* remove entries */
+	locker = g_rw_lock_writer_locker_new(&self->db_mutex);
+	g_return_val_if_fail(locker != NULL, FALSE);
+	rc = sqlite3_prepare_v2(self->db,
+				"INSERT INTO hsi_history (hsi_details, hsi_score)"
+				"VALUES (?1, ?2)",
+				-1,
+				&stmt,
+				NULL);
+	if (rc != SQLITE_OK) {
+		g_warning("write error %s", sqlite3_errmsg(self->db));
```

Comment:
is the warning really needed?  the same thing is returned as an error

Reply:
It is not necessary. I'll remove it.